frontend/src/features/settings/hooks/creditAccount.ts

```diff
@@ -0,0 +1,254 @@
+ * @returns Result of the create credit account action
+ */
+export const useCreateCreditAccountMutation = () => {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const { setSnackBar } = useContext(SnackBarContext);
+
+  return useMutation({
+    mutationFn: createCreditAccount,
+    onSuccess: (response) => {
+      queryClient.setQueryData(
+        ["credit-account", { companyId: getCompanyIdFromSession() }],
+        response.data,
+      );
+      setSnackBar({
+        showSnackbar: true,
+        setShowSnackbar: () => true,
+        message: "Credit Account Added",
+        alertType: "success",
+      });
+      queryClient.invalidateQueries({
+        queryKey: ["credit-account", { companyId: getCompanyIdFromSession() }],
+      });
+    },
+    onError: () => navigate(ERROR_ROUTES.UNEXPECTED),
+  });
+};
+
+/**
+ * Hook to add a user to a credit account
+ * @returns Result of the add user to credit account action
+ */
+export const useAddCreditAccountUserMutation = () => {
+  const { setSnackBar } = useContext(SnackBarContext);
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+
+  return useMutation({
+    mutationFn: (data: { creditAccountId: number; userData: CompanyProfile }) =>
+      addCreditAccountUser(data),
+    onSuccess: () => {
+      setSnackBar({
+        showSnackbar: true,
+        setShowSnackbar: () => true,
+        alertType: "success",
+        message: "Account User Added",
+      });
+
+      queryClient.invalidateQueries({
+        queryKey: ["credit-account", { companyId: getCompanyIdFromSession() }],
+      });
+    },
+    onError: () => navigate(ERROR_ROUTES.UNEXPECTED),
+  });
+};
+
+/**
+ * Hook to remove a user from a credit account
+ * @returns Result of the remove user from credit account action
+ */
+export const useRemoveCreditAccountUsersMutation = () => {
+  const navigate = useNavigate();
+  const { setSnackBar } = useContext(SnackBarContext);
+
+  return useMutation({
+    mutationFn: (data: { creditAccountId: number; companyIds: number[] }) =>
+      removeCreditAccountUsers(data),
+    onSuccess: () => {
+      setSnackBar({
+        showSnackbar: true,
+        setShowSnackbar: () => true,
+        message: "Account User(s) Removed",
+        alertType: "info",
+      });
+    },
+    onError: () => {
+      navigate(ERROR_ROUTES.UNEXPECTED);
+    },
+  });
+};
+
+/**
+ * Hook to hold/unhold a credit account.
+ * @returns Result of the hold credit account action
+ */
+export const useUpdateCreditAccountStatusMutation = () => {
+  const navigate = useNavigate();
+  const { setSnackBar } = useContext(SnackBarContext);
+
+  return useMutation({
+    mutationFn: (data: {
+      creditAccountId: number;
+      updateStatusData: UpdateStatusData;
+    }) => {
+      const {
+        creditAccountId,
+        updateStatusData: { updateStatusAction, reason },
+      } = data;
+
+      let status: CreditAccountStatusType;
```

Comment:
I feel like you could extract this status logic into a helper method of some sort (preferably put it in the same file as UPDATE_STATUS_ACTIONS), and simply reuse that method call here.